File: Data Structures & Algorithms/jump-game/submission-1.java
class Solution {
    public boolean canJump(int[] nums) {
        int i = nums.length - 2;
        int goal = nums.length - 1;
        while (i >= 0) {
            int curr = i + nums[i];
            if (curr >= goal) {
                goal = i;
            }
            i --;
        }
        return goal == 0;
    }
}

File: Data Structures & Algorithms/rearrange-array-elements-by-sign/submission-1.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        res = []

        positive, negative = 0, 0

        def next_pos():
            nonlocal positive
            while nums[positive] < 0 and positive < len(nums):
                positive += 1
        
        def next_neg():
            nonlocal negative
            while nums[negative] > 0 and negative < len(nums):
                negative += 1

        while positive < len(nums) and negative < len(nums):
            next_pos()
            next_neg()
            if positive >= len(nums) or negative >= len(nums):
                break
            res.append(nums[positive])
            res.append(nums[negative])
            positive += 1
            negative += 1
        
        return res